pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/bestpractices/JUnitTestsShouldIncludeAssertRule.java

```diff
@@ -173,29 +175,25 @@ private boolean isVerifyStatement(ASTStatementExpression expression) {
 
     private boolean isExpectStatement(ASTStatementExpression expression,
             Map<String, List<NameOccurrence>> expectables) {
-        
-        if (expression != null) {
-            
-            ASTPrimaryExpression pe = expression.getFirstChildOfType(ASTPrimaryExpression.class);
-            if (pe != null) {
-                Node name = pe.getFirstDescendantOfType(ASTName.class);
-                // case of an AllocationExpression
-                if (name == null) {
-                    return false;
-                }
-                
-                String img = name.getImage();
-                if (img.indexOf(".") == -1) {
-                    return false;
-                }
-                String varname = img.split("\\.")[0];
-
-                if (!expectables.containsKey(varname)) {
-                    return false;
+        ASTPrimaryExpression pe = expression.getFirstChildOfType(ASTPrimaryExpression.class);
+        if (pe != null) {
+            ASTPrimaryPrefix primaryPrefix = pe.getFirstChildOfType(ASTPrimaryPrefix.class);
+            Node name = pe.getFirstDescendantOfType(ASTName.class);
+            if (!primaryPrefix.usesThisModifier() && name != null) {
```

Comment:
We should probably one day consider `super` here too, but I don't think this would work right now... We'd probably have to explore supertypes via reflection.